gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -151,7 +156,20 @@ void onSignedOut() {
 
   private void onSignedIn() {
     hideProgressDialog();
-    navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+
+    /* Checking whether the user have accepted Terms & Conditions. */
+    if (localValueStore.isTermsAccepted()) {
+      navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+    } else {
+
+      /*  Checking here whether the user is coming from SignIn Fragment or StartUp Fragment
+       based on that redirecting user. */
+      if (signInProgressDialogVisibility.getValue() == null) {
+        navigator.navigate(StartupFragmentDirections.proceedToTermsScreen());
```

Comment:
if and else branch are doing the same thing - what am I missing here ?

Reply:
its checking whether user is coming from signin screen or startup screen using signInProgressDialogVisibility visibility